classes/script_metadata.php

```diff
@@ -136,14 +142,43 @@ function ($i) {
         );
 
         foreach ($parameters as $i => &$v) {
-            if (in_array($i, self::REDACTLIST)) {
+            if (in_array($i, self::get_redactable_param_names())) {
                 $v = '';
             }
         }
 
         return $parameters;
     }
 
+    /**
+     * Gets a list of parameter names to be redacted, combining those from settings with
+     * those builtin.
+     *
+     * @return string[]
+     */
+    public static function get_redactable_param_names(): array {
+        $setting = get_config('tool_excimer', 'redact_params');
+
+        // Strip C style comments.
+        $setting = preg_replace('!/\*.*?\*/!s', '', $setting);
+
+        $lines = explode(PHP_EOL, $setting);
+
+        $params = self::REDACTLIST;
+        foreach ($lines as $line) {
```

Comment:
A block comment explaining the purpose of this loop would be good. What do you think?

Reply:
The comment on line 167 addresses the loop. The loop is also self-explanetary, so a comment shouldn't be needed.

Reply:
Thanks for adding a comment. Also I realise that "block comment" might have refered to a larger comment. What I actually intended was a comment on the code block, which you have added.